code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -146,3 +146,53 @@
 	response_disarm = "gently pushes aside"
 	response_harm   = "splats"
 	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID
+
+
+/mob/living/simple_animal/mouse/blobinfected
+	maxHealth = 100
+	health = 100
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	var/cycles_alive = 0
+	var/cycles_limit = 30
+	var/has_burst = FALSE
+
+/mob/living/simple_animal/mouse/blobinfected/Life()
+	cycles_alive++
+	if(prob(20))
+		var/timeleft = (cycles_limit - cycles_alive) * 2
+		if(timeleft < 1)
+			burst(FALSE)
+		else
+			to_chat(src, "<span class='warning'>[timeleft] seconds until you burst, and become a blob...</span>")
+	return ..()
+
+/mob/living/simple_animal/mouse/blobinfected/death(gibbed)
+	burst(gibbed)
+	return ..(gibbed)
+
+/mob/living/simple_animal/mouse/blobinfected/proc/burst(gibbed)
+	if(has_burst)
+		return FALSE
+	var/turf/T = get_turf(src)
+	if(!is_station_level(T.z) || istype(T, /turf/space))
+		to_chat(src, "<span class='userdanger'>You feel ready to burst, but this isn't an appropriate place!  You must return to the station!</span>")

```

Comment:
what'd stop a taj from eating the mouse, then either leaving the level or taking a spacewalk

Reply:
The mouse will be player controlled, probably avoiding people, and likely in a vent 90% of the time. Still, it is in theory possible for a taj to quickly eat the mouse, without killing it, and then run into space before the short timer expires. They'd have to stay in space for as long as the mouse is inside them, though. Even a dead mouse inside them will gib them and turn into a core the second the taj gets back onto the station.

If you'd like me to put in some check that instantly turns them into a core if the mouse is eaten, let me know.

Reply:
Once the mouse dies when swallowed, it gets `qdel`'d. This would prevent it from re-firing and gibbing the taj.